### PR TITLE
Dummy store

### PR DESCRIPTION
**Pull request by vidartf**
This is an implementation of a *dummy* datastore:
- It declares an interface for a datastore based on those of lumino's datastore
- It implements a `Dummystore` class for that interface, which does not use any CRDT logic, and does not create or accept patches. This means it is a history-less, single-user datastore, with minimal overhead.
- This can be used by datastore clients to switch between using the CRDT capabilities or not, while still using the same interface for the data storage model.

### CODE REVIEW

File: packages/dummystore/lib/datastore.d.ts [hunk 1]
```diff
@@ -0,0 +1,170 @@
+import { IIterator } from '@lumino/algorithm';
```
Comment (saulshanabrook) [inline packages/dummystore/lib/datastore.d.ts]: Should `lib` be ignored in git?

Comment (vidartf) [reply packages/dummystore/lib/datastore.d.ts]: ah, probably. I'll clean it up

File: packages/dummystore/src/interface.ts [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+
+import {
+  Datastore, Schema, Record
+} from '@lumino/datastore';
+
+import { IIterable, IIterator } from '@lumino/algorithm';
+import { IDisposable } from '@lumino/disposable';
+import { ISignal } from '@lumino/signaling';
+
+export interface IDatastore extends IDisposable, IIterable<ITable<Schema>> {
```
Comment (saulshanabrook) [inline packages/dummystore/src/interface.ts]: Haven't looked in lumino datastore, but why don't we use `IDatastore` from there? It seems like where it should reside?

File: packages/dummystore/tests/build/bundle.test.js [hunk 3]
```diff
@@ -0,0 +1,527 @@
...==++s||0===n&&0===u)&&(i=this.mul(i,r[a]),s=0,a=0)):s=0}f=26}return i},x.prototype.convertTo=function(e){var t=e.umod(this.m);return t===e?t.clone():t},x.prototype.convertFrom=function(e){var t=e.clone();return t.red=null,t},o.mont=function(e){return new M(e)},i(M,x),M.prototype.convertTo=function(e){return this.imod(e.ushln(this.shift))},M.prototype.convertFrom=function(e){var t=this.imod(e.mul(this.rinv));return t.red=null,t},M.prototype.imul=function(e,t){if(e.isZero()||t.isZero())return e.words[0]=0,e.length=1,e;var r=e.imul(t),n=r.maskn(this.shift).mul(this.minv).imaskn(this.shift).mul(this.m),i=r.isub(n).iushrn(this.shift),o=i;return i.cmp(this.m)>=0?o=i.isub(this.m):i.cmpn(0)<0&&(o=i.iadd(this.m)),o._forceRed(this)},M.prototype.mul=function(e,t){if(e.isZero()||t.isZero())return new o(0)._forceRed(this);var r=e.mul(t),n=r.maskn(this.shift).mul(this.minv).imaskn(this.shift).mul(this.m),i=r.isub(n).iushrn(this.shift),a=i;return i.cmp(this.m)>=0?a=i.isub(this.m):i.cmpn(0)<0&&(a=i.iadd(this.m)),a._forceRed(this)},M.prototype.invm=function(e){return this.imod(e._invmp(this.m).mul(this.r2))._forceRed(this)}}(e,this)}).call(this,r(196)(e))},function(e,t,r){"use strict";(function(e){
```
Comment (saulshanabrook) [inline packages/dummystore/tests/build/bundle.test.js]: `tests/build` should probably be ignored too?

**Comment by saulshanabrook** [comment]
Looks good! Thanks Vidar. This will be helpful for testing!

**Comment by vidartf** [comment]
> This will be helpful for testing!

I was hoping it would be useful in practice as well.

File: packages/dummystore/src/interface.ts (hunk 2 above)
Comment (vidartf) [reply packages/dummystore/src/interface.ts]: Sure, we can upstream the interface definition to lumino once we've verified it is useful as it is.

**Comment by vidartf** [comment]
Did a rebase to clean out the committed outputs.

File: packages/dummystore/src/interface.ts (hunk 2 above)
Comment (saulshanabrook) [reply packages/dummystore/src/interface.ts]: And if you just change it to `export interface IDatastore extends Datastore {}` would that still work? Then we wouldn't have to duplicate all the methods and everything? (I know we usually do duplicate them in JupyterLab, but curious if we could avoid it).

Comment (vidartf) [reply packages/dummystore/src/interface.ts]: That would require that any implementors of the interface inherit from `Datastore`.

Comment (saulshanabrook) [reply packages/dummystore/src/interface.ts]: Ah, never new [that](https://www.typescriptlang.org/docs/handbook/interfaces.html#interfaces-extending-classes):

> This means that when you create an interface that extends a class with private or protected members, that interface type can only be implemented by that class or a subclass of it.

Wonder why that is...

**Pull request merged**